File: Vadim-Yakshigulov/Semester 2/Blackjack/src/main/kotlin/blackjack/classes/Deck.kt
package blackjack.classes

import blackjack.enums.CardFace
import blackjack.enums.CardSuit
import blackjack.interfaces.ICard
import blackjack.interfaces.IDeck

class Deck : IDeck {
    override val cards: List<ICard>
        get() = _cards
    private val _cards: MutableList<ICard> = mutableListOf()
    init {
        renew()
    }

    override fun renew() {
        _cards.clear()
        for (suit in CardSuit.values())
            for (face in CardFace.values()) {
                _cards.add(Card(suit, face))
            }
    }
}

File: StarcevMatvey/Term 2/Block_3/Task_1/BlackJack/Deck.cs
namespace BlackJack
{
    public class Deck
    {
        public List<Card> Cards { get; }

        public Deck()
        {
            Cards = new List<Card>();
            for (int i = 0; i < 52; i++)
            {
                Cards.Add(new Card(i % 4, i % 13 + 1));
            }
        }
    }
}
